blocks/monitoring/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""This part of the framework helps you monitor your model during training.
+
+:class:`MonitorChannels` describe quantities that should be monitored. They
+can either be Theano variables (e.g. the objective function, the weight
+norms, etc.) or they can be callable functions (e.g. if you want to use
+NumPy to calculate eigenvalues, or sample text from your language model).
+
+.. todo::
+
+   Monitored Theano variables might depend on inputs beyond those given to
+   the model. There should be a way of providing these to the monitoring
+   channels, collecting them, and compiling them as part of the Theano
+   function that performs monitoring.
+
+"""
+import logging
+from abc import ABCMeta, abstractmethod
+
+from blocks.utils import shared_expression, update_instance
+
+logger = logging.getLogger(__name__)
+
+
+class MonitorChannel(object):
```

Comment:
I know, it has the evaluators. I wanted to have them in VC to have a full view of things for later. Feel free to branch off and remove them, or I can push something without them.

You are correct about the assumption about datasources. It worked well in GH and seems natural to me -- the dataset yields dicts, we match stuff by name. It also solves your ..todo in #90 - the evaluator walks the graph to find all inputs (not just those of the model) and assumes that they will be present in what the dataset yields.